R/joyn-merge.R

```diff
@@ -231,6 +228,22 @@ joyn <- function(x,
   # the resulting table should have the same class as the x table.
   class_x <- class(x)
 
+  # ensure input names can be restored
+  byexp <- grep(pattern = "==?",
+                x       = by,
+                value   = TRUE)
+  xbynames <- trimws(gsub("([^=]+)(\\s*==?\\s*)([^=]+)",
+                          "\\1",
+                          byexp))
+  ybynames <- trimws(gsub("([^=]+)(\\s*==?\\s*)([^=]+)",
+                          "\\3",
+                          byexp))
+  ynames  <- copy(names(y))
+
+  # maintain name that is bound to original inputs
+  x_original <- x
+  y_original <- y
+
   # If match type is m:m we need to convert to data.table
   if (match_type == "m:m") {
```

Comment:
This is not necessary anymore given that we are using `collapse::join` for `m:m`, right?